lib/Raven/Client.php

```diff
@@ -233,35 +238,49 @@ class Client
      */
     protected $_shutdown_function_has_been_set;
 
-    public function __construct($options_or_dsn = null, $options = [])
+    /**
+     * @var Configuration The client configuration
+     */
+    protected $config;
+
+    /**
+     * Constructor.
+     *
+     * @param Configuration $config The client configuration
+     */
+    public function __construct(Configuration $config)
```

Comment:
I think the pattern of passing the args (vs another class) is fine here. Ruby is done via blocks so its not super relevant here (you dont need to intantiate a configuration object), and you can still pass the configuration to the 'Configuration' class constructor for validation/etc. The problem is that any native integration now nees to not only instantiate the Client class  (which is easy), but also pull in the config class, coerce config, and then pass that in.


e.g.

```
# pretend-framework.yml
exceptionHandler: \Sentry\Client
exceptionHandlerArgs: ...
```

If we're REALLY concerned, we can simply support both. If it's a string, its the DSN, if its Configuration [...], if its an array, its passed into Configuration.